server/src/gameplay/game_event_handler.rs

```diff
@@ -285,5 +328,10 @@ pub fn update_game_state(game_state: &mut GameState, event: Event) -> Result<(),
             game_state.spent_strategy_cards = HashSet::new();
         }
     }
+
+    // TODO: maybe not recalculate this all the time?
```

Comment:
Yeah this would be nice to handle only when we update score but I guess that could be in quite a few places.

Reply:
my thoughts exactly, we can optimize it if necessary